samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/shoppingcarts/removingproductitem/RemoveProductItemFromShoppingCart.java

```diff
@@ -0,0 +1,41 @@
+package io.eventdriven.ecommerce.shoppingcarts.removingproductitem;
+
+import io.eventdriven.ecommerce.shoppingcarts.Events;
+import io.eventdriven.ecommerce.shoppingcarts.ShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.productitems.PricedProductItem;
+
+import java.util.UUID;
+
+public record RemoveProductItemFromShoppingCart(
```

Comment:
Is it a command?
In my [repository](https://github.com/CamilYed/readable-tests-by-example/blob/master/src/main/java/tech/allegro/blog/vinyl/shop/order/application/OrderPaymentHandler.java#L53) I have named such objects explicitly adding a suffix.

Reply:
I don't use suffixes or prefixes. I think proper naming is enough, e.g. for commands using an imperative tense, for events past tense. What benefit would we have from suffixes?

Regarding architecture style, it's CQRS or sliced architecture, so code is sliced per specific business operations. The focus is on modelling business operations and their intent (command or queries). I'm trying to cut as many layers and ceremonies as possible (I'm not sure if I'm succeeding 😉 ).

Reply:
As far as I see it's some form of vertical slice. @oskardudycz wrote about it here:
https://event-driven.io/pl/how_to_slice_the_codebase_effectively/